commit 4bc889c01ce68475ce36f9c67b9a445a6c0218d9
Author: Noah Goldstein <goldstein.w.n@gmail.com>
Date:   Wed Jun 22 16:34:42 2022 -0700

    stdlib: Fixup mbstowcs NULL __dst handling. [BZ #29279]
    
    commit 464d189b9622932a75302290625de84931656ec0 (origin/master, origin/HEAD)
    Author: Noah Goldstein <goldstein.w.n@gmail.com>
    Date:   Wed Jun 22 08:24:21 2022 -0700
    
        stdlib: Remove attr_write from mbstows if dst is NULL [BZ: 29265]
    
    Incorrectly called `__mbstowcs_chk` in the NULL __dst case which is
    incorrect as in the NULL __dst case we are explicitly skipping
    the objsize checks.
    
    As well, remove the `__always_inline` attribute which exists in
    `__fortify_function`.
    Reviewed-by: Siddhesh Poyarekar <siddhesh@sourceware.org>
    
    (cherry picked from commit 220b83d83d32aa9e6f5659e2fa2a63a0024c3e4a)

diff --git a/stdlib/bits/stdlib.h b/stdlib/bits/stdlib.h
index f0918d5d798a6ac4..ac9badc81f3990c5 100644
--- a/stdlib/bits/stdlib.h
+++ b/stdlib/bits/stdlib.h
@@ -96,10 +96,10 @@ extern size_t __mbstowcs_chk (wchar_t *__restrict __dst,
 			      const char *__restrict __src,
 			      size_t __len, size_t __dstlen) __THROW
     __attr_access ((__write_only__, 1, 3)) __attr_access ((__read_only__, 2));
-extern size_t __REDIRECT_NTH (__mbstowcs_chk_nulldst,
+extern size_t __REDIRECT_NTH (__mbstowcs_nulldst,
 			      (wchar_t *__restrict __dst,
 			       const char *__restrict __src,
-			       size_t __len), mbstowcs_chk)
+			       size_t __len), mbstowcs)
     __attr_access ((__read_only__, 2));
 extern size_t __REDIRECT_NTH (__mbstowcs_alias,
 			      (wchar_t *__restrict __dst,
@@ -113,12 +113,12 @@ extern size_t __REDIRECT_NTH (__mbstowcs_chk_warn,
      __warnattr ("mbstowcs called with dst buffer smaller than len "
 		 "* sizeof (wchar_t)");
 
-__always_inline __fortify_function size_t
+__fortify_function size_t
 __NTH (mbstowcs (wchar_t *__restrict __dst, const char *__restrict __src,
 		 size_t __len))
 {
   if (__builtin_constant_p (__dst == NULL) && __dst == NULL)
-    return __mbstowcs_chk_nulldst (__dst, __src, __len);
+    return __mbstowcs_nulldst (__dst, __src, __len);
   else
     return __glibc_fortify_n (mbstowcs, __len, sizeof (wchar_t),
 			      __glibc_objsize (__dst), __dst, __src, __len);
